synapse/storage/databases/main/end_to_end_keys.py

```diff
@@ -726,6 +797,23 @@ def _claim_e2e_one_time_keys(txn):
                 self._invalidate_cache_and_stream(
                     txn, self.count_e2e_one_time_keys, (user_id, device_id)
                 )
+            # mark fallback keys as used
+            for user_id, device_id, algorithm, key_id in used_fallbacks:
+                self.db_pool.simple_update_txn(
+                    txn,
+                    "e2e_fallback_keys_json",
+                    {
+                        "user_id": user_id,
+                        "device_id": device_id,
+                        "algorithm": algorithm,
+                        "key_id": key_id,
+                    },
+                    {"used": True},
```

Comment:
We track that these were used so that the client can replace them next time they're online, correct?

Reply:
yup